### DISCUSSION TITLE
LCM - please help me to understand

### DISCUSSION THREAD

**Discussion by cgidesign-de** [vladmandic/sdnext#3299] [Q&A] [answered]
I googled but struggle to find a good overview:

There is LCM to speed up image generation.
I like to use it with Diffuser Backend - SDXL.

I found:

A: LCM Lora
B: LCM Checkpoints
C: LCM sampler

What I am unsure about is the relation between those:

I.e.
Can I use any SDXL 1.0 checkpoint with LCM sampler?
I tried but get bad results.

Must I use LCM Checkpoints?
I tried those and get usable results.

Can I use any SDXL 1.0 checkpoint with LCM Lora?
I tried but get bad results.

Sorry for asking but I am confused.

**Comment by VeteranXT** [comment, marked as answer]
your question : I.e.
Can I use any SDXL 1.0 checkpoint with LCM sampler
Yes, but that is sampler. Output can be bad.

Must I use LCM Checkpoints? 

Yes and not. There is LCM Lora that you can use to apply to all of same types of model
Aka: SD1.5 LCM Lora to SD1.5 checkpoint and same goes for SDXL lora to SDXL model.
sing SDXL LCM model you do not need lora. same goes to SD 1.5.

Can I use any SDXL 1.0 checkpoint with LCM Lora?
Yes, But do not mix model versions 1.5 with XL when using loras.

For best results to keep cfg scale to default 7 is to download hyper
https://huggingface.co/ByteDance/Hyper-SD/tree/main
Expectantly Hyper 8 Steps CFG for both XL and 1.5 
Others you have to keep CFG scale low.
Reason why you can use CFG from 1-20 while with LCM your CFG scale must be between 1-2.5.

**Comment by cgidesign-de** [comment]
Thanks for the information.
With the hyper loras it seems to work a bit better but only if I keep cfg a 1.
But all in all the "normal" checkpoints without LCM produce way better results for me.

Do I need to change VAE model or settings as well for LCM?

**Reply by VeteranXT** [reply]
Nope. VAE is just way to encode/decode.
There is SD1.5 : Hyper-SD15-8steps-CFG-lora.safetensors
for SDXL :  Hyper-SDXL-12steps-CFG-lora.safetensors and Hyper-SDXL-8steps-CFG-lora.safetensors
With this one you can use CFG freely. Others can be form 1 to 1.5-2 is max

**Comment by cgidesign-de** [comment]
Again, thanks @VeteranXT
I am getting usable results with LCM now :-)

**Reply by VeteranXT** [reply]
You're welcome!
